src/components/search-panel/SearchPanel.tsx

```diff
@@ -0,0 +1,224 @@
+		setFormType(type => (type === 'event' ? 'message' : 'event'));
+	};
+
+	const [formState, setFormState] = useSetState<SearchPanelState>(defaultFormState);
+	const [eventFilter, setEventFilter] = useSetState<EventFilterState>();
+	const [messagesFilter, setMessagesFilter] = useSetState<MessageFilterState>();
+
+	const [results, setResults] = useState<EventAction[]>([]);
+	const [currentlyLaunchedChannel, setCurrentlyLaunchedChannel] = useState<EventSource | null>(
+		null,
+	);
+
+	useEffect(() => {
+		setEventFilter(getDefaultFilterState(eventFilterInfo));
+		setMessagesFilter(getDefaultFilterState(messagesFilterInfo));
+	}, [eventFilterInfo, messagesFilterInfo]);
+
+	const form = { formType, state: formState, setState: setFormState };
+	const filters = useMemo(
+		() =>
+			formType === 'event'
+				? { info: eventFilterInfo, state: eventFilter, setState: setEventFilter }
+				: { info: messagesFilterInfo, state: messagesFilter, setState: setMessagesFilter },
+		[formType, eventFilter, messagesFilter],
+	);
+
+	const launchChannel = useCallback(() => {
+		const {
+			startTimestamp,
+			searchDirection,
+			resultCountLimit,
+			timeLimit,
+			parentEvent,
+			stream,
+		} = formState;
+
+		const currentFilters = formType === 'event' ? eventFilterInfo : messagesFilterInfo;
+
+		const filterParams = formType === 'event' ? eventFilter : messagesFilter;
+
+		function getFilter<T extends keyof FilterState>(name: T) {
+			return filterParams[name];
+		}
+
+		const filtersToAdd = currentFilters
+			.filter((info: SSEFilterInfo) => {
+				const values = getFilter(info.name).values;
+				return values.length !== 0;
+			})
+			.map((info: SSEFilterInfo) => info.name);
+
+		const filterValues = filtersToAdd.map(filter => {
+			const values = getFilter(filter).values;
+			return [`${filter}-values`, values];
+		});
+
+		const filterInclusion = filtersToAdd.map(filter => {
+			const negative = getFilter(filter).negative;
+			return negative ? [`${filter}-negative`, negative] : [];
+		});
+
+		const params = {
+			startTimestamp,
+			searchDirection,
+			resultCountLimit,
+			timeLimit,
+			filters: filtersToAdd,
+			...Object.fromEntries([...filterValues, ...filterInclusion]),
+		};
+
+		const queryParams = formType === 'event' ? { ...params, parentEvent } : { ...params, stream };
+
+		const channel = sseApi.getEventSource({
+			type: formType,
+			queryParams,
+			listener: (ev: Event | MessageEvent) => {
+				const data = (ev as MessageEvent).data;
+				setResults(res => [...res, JSON.parse(data)]);
+			},
+			onClose: () => {
+				setCurrentlyLaunchedChannel(null);
+			},
+			onOpen: () => {
+				setResults([]);
+			},
+		});
+
+		setCurrentlyLaunchedChannel(channel);
+	}, [formType, formState, eventFilter, messagesFilter]);
+
+	const stopChannel = useCallback(() => {
+		currentlyLaunchedChannel?.close();
+		setCurrentlyLaunchedChannel(null);
+	}, [currentlyLaunchedChannel]);
+
+	const currentResults: Result[] = React.useMemo(() => {
```

Comment:
I dont think it very usefull to reiterate over values just to add a type and then inside of SearchPanelResults you checking type again. Just pass results as they are